module/VuFind/src/VuFind/Controller/AbstractSolrSearch.php

```diff
@@ -89,6 +89,18 @@ public function advancedAction()
         return $view;
     }
 
+    /**
+     * Get proper options file for search class
+     *
+     * @return \VuFind\Search\Base\Options
+     */
+    public function getOptionsForClass(): \VuFind\Search\Base\Options
```

Comment:
We're doing exactly the same thing [here](https://github.com/vufind-org/vufind/blob/9e56d78621677947fa961d6a2547de17d6f4350c/module/VuFind/src/VuFind/Controller/AbstractSearch.php#L97) -- maybe it's worth moving this method up to AbstractSearch and using it there so it can be more easily shared across more controllers.